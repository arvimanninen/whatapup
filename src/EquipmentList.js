import React from 'react';

class EquipmentList extends React.Component {
    render(){
        const category = this.props.category;
        const items = category.items.map((item) => 
            <button key={item.key} 
            onClick={() => this.props.handleItemClick(category.key, item.key)}>{item.name}</button>);
        return(
            <div>
                <h3>{category.name}</h3>
                <div>{items}</div>
                <button onClick={() => this.props.handleViewChange("MainMenu")}>Back</button>
            </div>
        );
    }
}

export default EquipmentList;