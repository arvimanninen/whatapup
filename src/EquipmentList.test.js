import React from 'react';
import renderer from 'react-test-renderer';
import EquipmentList from './EquipmentList';
import CATEGORIES from './index';

test('Renders EquipmentList', () => {
  const component = renderer.create(<EquipmentList category={CATEGORIES[0]}></EquipmentList>,);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});