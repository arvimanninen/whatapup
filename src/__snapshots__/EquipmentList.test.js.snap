// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders EquipmentList 1`] = `
<div>
  <h3>
    Category A
  </h3>
  <div>
    <button
      onClick={[Function]}
    >
      Item a1
    </button>
    <button
      onClick={[Function]}
    >
      Item a2
    </button>
    <button
      onClick={[Function]}
    >
      Item a3
    </button>
  </div>
  <button
    onClick={[Function]}
  >
    Back
  </button>
</div>
`;
